### ISSUE TITLE
http-route does not support Consul cluster peering

### ISSUE THREAD

**Issue by vvarga007** [hashicorp/consul-api-gateway#593]
http-route configuration entry does not support Consul cluster peering.

### Feature Description
I would like to use the API gateway for imported services. Imported services via cluster peering.
The [documentation](https://developer.hashicorp.com/consul/docs/connect/gateways/api-gateway/configuration/http-route)
does not say anything about cluster peering.

Adding "Peer" directive would be nice, something like this: 
```yaml
Rules = [
      {
        Matches = [
          {
            Path = {
              Match = "prefix"
              Value = "/"
            }
          }
        ]
        Services = [
          {
            Name = "hashicups-frontend"
            Peer = "cluster-02"
          },
        ]
      },
]
```

**Comment by nathancoleman** [comment]
Hi @vvarga007 :wave:

There are Kubernetes-specific instructions [here](https://developer.hashicorp.com/consul/docs/api-gateway/usage/route-to-peered-services) for routing to a service in a peer. It requires the use of the `MeshService` CRD which allows you to specify which peer the service is imported from. Let me know if this gets you up and running 🎉

**Comment by vvarga007** [comment]
@nathancoleman I don't use Kubernetes. I use Nomad.

**Comment by nathancoleman** [comment]
I believe you can accomplish this in Consul today using a [Service Resolver configuration entry](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-resolver) and, more specifically, its ability to [redirect to a peer](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-resolver#redirect-peer). If `hashicups-frontend` is a service located in a peered cluster named `my-peer`, you would need the additional configuration that would look something like this:

```hcl
Kind = "service-resolver"
Name = "hashicups-frontend"
Redirect {
  Service = "hashicups-frontend"
  Peer = "my-peer"
}
```

and then your `http-route` would remain
```hcl
Rules = [
      {
        Matches = [
          {
            Path = {
              Match = "prefix"
              Value = "/"
            }
          }
        ]
        Services = [
          {
            Name = "hashicups-frontend"
          },
        ]
      },
]
```

Let me know if this is helpful for you or not. If you like, I can put together a config bundle of what I'm describing -- just let me know 😃

**Comment by vvarga007** [comment]
Yes, it is possible with the service resolver, but I have to create a fake/virtual service.
So let's say hashicups is running in DC1, and I want to spin up an ingress gw/api gw in DC2. The hashicups service is exported in DC1 and imported in DC2. To be able to set up a service resolver in DC2, I have to create a fake/virtual service. It would be great to do this without a virtual service and target a service through a peering connection.

**Comment by nathancoleman** [comment]
Ah, I see your point. So this issue isn't so much that you can't accomplish the thing but more that it could be easier with some UX tweaks, right?

**Comment by vvarga007** [comment]
Yes, it is more like a feature request.

**Comment by nathancoleman** [comment]
@vvarga007 excellent! This repo will soon be marked deprecated as the api-gateway functionality has been integrated into [hashicorp/consul](https://github.com/hashicorp/consul) and [hashicorp/consul-k8s](https://github.com/hashicorp/consul-k8s). Nothing required on your end, but a heads up that I expect this feature request to be migrated to the consul repo in the near-ish future.
